### PR TITLE
Add AE Test Part2

### PR DESCRIPTION
**Pull request by karinazhou**
- Add SqlNullValuesTest 
- Modify the test setting minorly

### CODE REVIEW

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+            }
+        }
+
+        [ConditionalTheory(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringSetupForAE))]
+        [ClassData(typeof(NullValueTestsData))]
+        public void NullValueTests(string connString, ConnectionStringSetting connStringSetting, SqlCommandColumnEncryptionSetting commandSetting, ReturnValueSetting nullReturnValue)
+        {
+            switch (connStringSetting)
+            {
+                case ConnectionStringSetting.Enabled:
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: "ConnectionStringSetting" can be renamed to "ConnStringColumnEncryptionSetting"?
(Looking at SqlCommandColumnEncryptionSetting)

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                    connString += "; Column Encryption Setting=Disabled";
+                    break;
+            }
+
+            using (SqlConnection sqlConn = new SqlConnection(connString))
+            {
+                sqlConn.Open();
+
+                // Create a command similarly
+                SqlCommand cmd = new SqlCommand(String.Format("SELECT c1 FROM [{0}] ORDER BY c2 ASC", tableName),
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: If not using `using`, please add `cmd.close();` and `reader.close()` in `finally` of a try-finally block.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            }
+        }
+
+        private void DropFunctions(SqlConnection sqlConnection)
+        {
+            string[] funcNames = {UdfName, UdfNameNotNull};
+
+            foreach (string funcName in funcNames)
+            {
+                using (SqlCommand cmd = new SqlCommand(string.Format("IF EXISTS (SELECT * FROM sys.objects WHERE name = '{0}') \n DROP FUNCTION {0}", funcName), sqlConnection))
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: Please move this re-usable code in a new `DropFunction(SqlConnection con)` API in DataTestUtility.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+        public void Dispose()
+        {
+            foreach (string connStrAE in DataTestUtility.AEConnStringsSetup)
+            {
+                using (SqlConnection sqlConnection = new SqlConnection(connStrAE))
+                {
+                    sqlConnection.Open();
+                    Table.DeleteData(fixture.SqlNullValuesTable.Name, sqlConnection);
+                    DropFunctions(sqlConnection);
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: Please call `Drop()` on "Table" `fixture.SqlNullValuesTable` in the end.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        Disabled
+    };
+
+    public enum ReturnValueSetting
+    {
+        NotNull,
+        Null
+    };
+
+}
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: New line EOF

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/SqlNullValuesTable.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            
+            using (SqlCommand command = sqlConnection.CreateCommand())
+            {
+                command.CommandText = sql;
+                command.ExecuteNonQuery();
+            }
+        }
+    }
+
+
```
Comment (cheenamalhotra) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/SqlNullValuesTable.cs]: Remove extra lines.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 1 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: Updated

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 4 above)
Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: The fixture should already be dropping the table when it is disposed by xunit. This just looks like it is cleaning up the additional data the test is writing to it.

Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: This table will be dropped when the tests are finished. Actually, all the tables created by SQLSetupStrategy will be dropped automatically. 
See [public void Dispose()](https://github.com/dotnet/SqlClient/blob/master/src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/SQLSetupStrategy.cs)

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/SqlNullValuesTable.cs (hunk 6 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/TestFixtures/Setup/SqlNullValuesTable.cs]: Done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 4 above)
Comment (cheenamalhotra) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: Yes, that's right. Thanks for reminding!

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 2 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: 'SqlCommand' does not contain a definition for 'Close'. I will use `using` block instead.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 3 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: Done.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 5 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: I tried locally but it looks like it is still there. How can I remove this from editor?

Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: You just need to add a return at the end of line 203.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+                    {
+                        Assert.True(reader.Read(), "No rows fetched first time");
+
+                        // Depending on the various flags verify the results
+                        object value1 = reader.GetProviderSpecificValue(0);
+
+                        // Read next row
+                        Assert.True(reader.Read(), "No rows fetched second time");
+                        object value2 = reader.GetProviderSpecificValue(0);
+
```
Comment (David-Engel) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: You can now close `}` the reader's using. You don't need lines 96-98 anymore. You could even close `}` the cmd's using here, too. cmd2's using can be moved down to before line 100, too, and then closed before the switch statement.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 5 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: Oh, OK. Done :)

File: src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs (hunk 7 above)
Comment (karinazhou) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/AlwaysEncrypted/SqlNullValues.cs]: It sounds reasonable. I will change it and also move some of the variables' declaration out of `using` block.

**Pull request merged**